ext/mysqli/tests/gh15432.phpt

```diff
@@ -0,0 +1,63 @@
+--TEST--
+GH-15432 PHP crashes when processing a MySQL DB query with a new Vector format introduced in MySQL 9.0
+--EXTENSIONS--
+mysqli
+--SKIPIF--
+<?php
+require_once 'connect.inc';
+if (!$link = @my_mysqli_connect($host, $user, $passwd, $db, $port, $socket)) {
+    die(sprintf("skip Can't connect to MySQL Server - [%d] %s", mysqli_connect_errno(), mysqli_connect_error()));
+}
+
+if ($link->server_version < 90000 || $link->server_version >= 10_00_00) {
+    die("skip MySQL 9.0.0+ needed");
+}
```

Comment:
Not really something that should be addressed in this PR, but still somewhat related:

What do you think about having something like `my_mysqli_try_connect()` which would `die()` on connection failure, but otherwise return the connection "link"? That could make this (and likely a couple of other tests) more efficient and more readable, e.g.
````php
require_once 'connect.inc';
$link = my_mysqli_try_connect($host, $user, $passwd, $db, $port, $socket);
if ($link->server_version < 90000 || $link->server_version >= 10_00_00) {
    die("skip MySQL 9.0.0+ needed");
}

Reply:
ohh, we have something like this already. Girgias made it some time ago. Check out ext\mysqli\tests\test_setup\test_helpers.inc

Reply:
For the record, see #15470.